src/components/Books.js

```diff
@@ -1,35 +1,42 @@
+import { useDispatch, useSelector } from 'react-redux';
+import { useState } from 'react';
+import { v4 as uuidv4 } from 'uuid';
+import BookItem from './BookItem';
+import { addBook } from '../redux/books/booksSlice';
+/* eslint-disable react/jsx-props-no-spreading */
 const Books = () => {
-  const bookList = [
-    {
-      id: 1,
-      title: 'The Loard',
-    },
-    {
-      id: 2,
-      title: 'Be Yourself!',
-    },
-    {
-      id: 3,
-      title: 'God and Human',
-    },
-  ];
+  const [title, setTitle] = useState('');
+  const [author, setAuthor] = useState('');
 
+  const { books } = useSelector((state) => state.books);
+  const dispatch = useDispatch();
+  const addBookHandler = () => {
+    const newBook = {
+      id: uuidv4(),
+      title,
+      author,
+    };
+    dispatch(addBook(newBook));
+  };
   return (
     <>
       <div className="booksList">
         <h5>Book&apos;s List:</h5>
-        {bookList.map((element) => (
-          <div className="bookState" key={element.id}>
-            <h3>{element.title}</h3>
-            <button type="button" className="delete">Delete</button>
-          </div>
+        {books.map((element) => (
+          <BookItem key={element.item_id} {...element} />
         ))}
       </div>
       <form action="submit">
         <h5>Add new Book:</h5>
-        <input type="text" placeholder="Book Title..." />
-        <input type="text" placeholder="Category" />
-        <button type="button" className="addBtn">Add Book</button>
+        <input type="text" placeholder="Book Title..." onChange={(e) => setTitle(e.target.value)} />
+        <input type="text" placeholder="Author" onChange={(e) => setAuthor(e.target.value)} />
+        <button
+          type="button"
+          className="addBtn"
+          onClick={addBookHandler}
+        >
+          Add Book
+        </button>
```

Comment:
- [ ] Please 🙏🏻 adhere to the project requirements, according to which you should add the `onClick` event on the **Button** component, which will help to add the new book into the `redux` store using the reducer. Right now you are using the `onClick` event but using `jsx` button tag, not the component, which is fine but not fulfilling the requirement. You should create a Button.js file for creating the component and then import that component and add the `click` event to make things work as it working right now.

### Requirement
![Screenshot from 2023-08-23 21-53-57](https://github.com/MohamedRoshdy2021/Book-store/assets/66476329/c7ad4113-ea5e-4e3f-9794-d95fa9e084dd)


### Caution ⚠️ 
**Please you need to replace your form submit button with the <Button onClick={Your_Event_Callback}> Add Book </Button>**.